### ISSUE TITLE
Long press menu > edit playlist on created playlists not working

### ISSUE THREAD

**Issue by KCreek1** [riverscuomo/public-bug-hunt#105]
**Describe the bug**
Long press > edit playlist on a created playlist in home or library screen opens whatever playlist was last selected not necessarily the created playlist. 

**To Reproduce**
1. Go to either home or library screen
2. Long press on a created playlist
3. Select edit playlist
4. See error - will open whatever playlist was last selected, not necessarily the created playlist. 

**Expected behavior**
When selecting edit playlist it should open the created playlist that you want to edit. 

**Screenshots**

https://github.com/riverscuomo/public-bug-hunt/assets/144278926/89bfcaac-f6b4-429c-af09-ae7f75277003

https://github.com/riverscuomo/public-bug-hunt/assets/144278926/a801073d-8a93-4a43-b342-e62894e8151c

**Smartphone (please complete the following information):**
 - Device: iPhone SE 2nd gen
 - OS: iOS 17.3
 - Version weezify iOS 203 (1.15.0)

**Additional context**
If playlist is opened and edit playlist is selected - this function works as expected.

**Comment by KCreek1** [comment]
On iPhone in weezify ios 215 (1.17.2)

The edit playlist option still opens whatever playlist you previously opened. 

However, there is a slight difference on this build: previously, if you tried to edit the playlist without first opening another, you would receive an "empty playlist" message. 

Now, if you take the same action, it shows one track: the one that is currently loaded in the player bar. See video below. 

https://github.com/riverscuomo/public-bug-hunt/assets/144278926/d1d497ed-f6f4-4666-8c2f-f415c6a32456

**Comment by Mel01000001** [comment]
Android build 215 (1.17.2),

Same occurs as described by @KCreek1 above re opening previously viewed playlist when long press to edit playlist and when trying to edit playlist when first opening app, the song in the player bar is the one listed in the playlist to be edited.

**Comment by riverscuomo** [comment]
Is this still a problem? I'm not able to recreate it.

**Comment by KCreek1** [comment]
@riverscuomo yes. It still occurs. Whatever playlist I open before I try to edit is the tracklist that appears. If I haven't opened a playlist yet, it lists the song shown in the player bar. 

In the video, I go straight to editing and you can see it's just the song that's in the player bar. Then I open Green and then try to edit my playlist and it shows Green tracklist. 
Finally, I open my playlist and when I go back and try the edit function - it shows the correct tracklist. 

https://github.com/riverscuomo/public-bug-hunt/assets/144278926/bcb111fc-d4ce-447d-8db1-301f9b23c3b6

**Comment by Mel01000001** [comment]
Yes.  It still happens on Android too.  Just a brief example here:


https://github.com/riverscuomo/public-bug-hunt/assets/144282154/560bbbf9-a7d5-47ee-9818-66d5694f1648

**Comment by KCreek1** [comment]
On weezify ios 226 (1.19.2)

I now have 2 different results:

Result 1:
1.) open app
2.) open any playlist other than the one you want to edit (in this case best of green) 
3.) return to library
4.) long press on a created playlist
5.) notice tracklist is from the initial playlist selected

https://github.com/riverscuomo/public-bug-hunt/assets/144278926/df058f98-dbae-492b-b1ff-c8b4576f42c3



Result 2a:
1.) open app
2.) long press on created playlist
3.) notice tracklist is correct. 

https://github.com/riverscuomo/public-bug-hunt/assets/144278926/9ccc4fd5-7035-4826-8835-0ddecbfbe9b5


Result 2b:
1.) open app
2.) open any playlist other than the one you want to edit (in this case best of green)
3.) return to library
4.) long press on a created playlist
5.) notice tracklist is from the initial playlist selected
6.) return to library
7.) open created playlist
8.) return to library
9.) long press on created playlist
10.) notice tracklist is correct

https://github.com/riverscuomo/public-bug-hunt/assets/144278926/fb14aa63-74cd-4cc4-91d9-78d9312fbe31



It seem that now the only time it doesn't work as expected is if another playlist is opened first (result 1). But then can be rectified by following the steps in 2b. 

In both 2a and 2b - once the correct tracklist displays then the long press > edit functions works as expected as long as the app is open, even if you open other playlists. You can go back and long press > edit and it shows[]() the correct tracks.
